### ISSUE TITLE
Issues if array has Getter

### ISSUE THREAD

**Issue by dadiborn** [wallabyjs/quokka#478]
### Issue description or question
If Array contains Getters/Setters, when you try to expand it inside Value Explorer: 
1 - Error in Value Explorer: Cannot delete property '1'...
 2 - in 'pseudo' array new value will be added  for no reason
3 - getting value will be undefined
![quokka_bug](https://user-images.githubusercontent.com/36768799/81890198-0e995780-95ae-11ea-9262-8fc3a11e00c9.jpg)

### Sample code
```javascript
let size = 2;
const real = [1,2,3,4];
const pseudo = Array(size);
for (let i = 0; i < size; i++) {
    Object.defineProperty(pseudo, i, {
        get(){
            return real[i]
        },
        set(value){
            return real[i] = value;
        }
    });
}

real;//?
pseudo;//?
pseudo[0]//?
```
Results are same on VSCode and Webstorm.
### OS name and version
Windows 10

**Comment by smcenlly** [comment]
Thanks for reporting the problem. We're not quite sure exactly what's going on yet, but we can duplicate the issue outside of Quokka using:

```javascript
let size = 2;
const real = [1,2,3,4];
const pseudo = Array(size);
for (let i = 0; i < size; i++) {
    Object.defineProperty(pseudo, i, {
        get(){
            return real[i]
        },
        set(value){
            return real[i] = value;
        }
    });
}

pseudo.splice(0, pseudo.length);
```

Node.js returns:
```
pseudo.splice(0, pseudo.length);
       ^

TypeError: Cannot delete property '1' of [object Array]
    at Array.splice (<anonymous>)
    at Object.<anonymous> (C:\Temp\facebook-jest\examples\mongodb\index.js:15:8)
    at Module._compile (internal/modules/cjs/loader.js:1158:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1178:10)
    at Module.load (internal/modules/cjs/loader.js:1002:32)
    at Function.Module._load (internal/modules/cjs/loader.js:901:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:74:12)
    at internal/main/run_main_module.js:18:47
```

Interestingly, if we set the array before it's defined, the code runs successfully:

```diff
  let size = 2;
  const real = [1,2,3,4];
  const pseudo = Array(size);
+ pseudo[0] = undefined;
+ pseudo[1] = undefined;
   for (let i = 0; i < size; i++) {
       Object.defineProperty(pseudo, i, {
           get(){
               return real[i]
           },
           set(value){
               return real[i] = value;
           }
       });
   }

   pseudo.splice(0, pseudo.length);
```

---

We're going to investigate further as to why this is occurring but it may be that defining the property on the array as you are doing is not supported by node (at least not without breaking some functions of the Array).

**Comment by dadiborn** [comment]
Then I will say Thank You Too twice!
First: Need to find another approach to get value :)
I have flat array and wanted to get part of it as array where changes will affect original values in flat array. (If you have any idea that would be off topic but super appreciated)
Second: Thank you for that super fantastic products with astral and hard to type titles (Wallaby...Quokka) - Absolute winner in most useful extensions ever made!

**Comment by smcenlly** [comment]
We've found that this is caused by `Configurable` attribute specifying that the property cannot be deleted from the object (mentioned in [defineProperty docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty)), which is why `slice()` is failing. By default, the configurable property is `false`, which is what is causing Wallaby and Array.slice() to fail. 

As a short term workaround, you may set configurable to `true` to get around the issue. Unless you very specifically don't want the array object itself being modified, this should let you do what you want to do. We modify the value as we report it in value explorer to prevent against large Arrays and other optimizations that are important to us.

We're continuing to investigate a proper fix for this that will allow you to use configurable `true` and still have Wallaby correctly display your values.

**Comment by dadiborn** [comment]
Thank you! Yes I just found that while smoking and came to response about configurable default to false in  defineProperty but you was faster :) , Will try it now. Thanks for so detailed explanation!

**Comment by smcenlly** [comment]
This has been fixed and is available in the latest version of Wallaby core, v1.0.893.
